byron/ledger/executable-spec/src/Ledger/UTxO/Generators.hs

```diff
@@ -80,8 +80,9 @@ genTxFromUTxO addrs txfee utxo = subtractFees txfee $ uncurry Tx <$> Gen.filter
   )
 
 subtractFees :: (Tx -> Lovelace) -> Gen Tx -> Gen Tx
-subtractFees txfee = fmap subtractFees'
-  . Gen.filter (\tx -> sum (value <$> outputs tx) >= txfee tx)
+subtractFees txfee = Gen.filter (not . null . outputs) <$>
+  fmap subtractFees'
```

Comment:
Maybe we shouldn't mix `fmap` and `<$>`. Can't we just use one of them?

Reply:
Also I I was thinking whether it'd make sense to perform the filtering only once: we don't care about the condition `\tx -> sum (value <$> outputs tx) >= txfee tx` since this will result in an empty list of outputs anyway.

Reply:
And if we need `fmap . fmap` the symbolic version is `<<$>>` in `Universum`.